packages/nuxt-module/plugins/domain.js

```diff
@@ -140,7 +140,7 @@ export default async ({ app, route, req }, inject) => {
        */
 
       const hasPrefixPath = (domainConfig) => {
-        return domainConfig.pathPrefix !== "/";
+        return domainConfig?.pathPrefix !== "/";
```

Comment:
If domainConfig is `undefined` the result is `true`. maybe it would be safer if we return false if domainConfig is not set to have false instead of true in this case? WDYT

Reply:
I don't know how this can happen, because every route should have the domainConfig now...
Give me some minutes to think this through

Reply:
The error can come from two places essentially: 
1) https://github.com/vuestorefront/shopware-pwa/blob/258e6d51193ff2f8eae70b53af62d265fd3e4141/packages/nuxt-module/plugins/domain.js#L149 
2) https://github.com/vuestorefront/shopware-pwa/blob/258e6d51193ff2f8eae70b53af62d265fd3e4141/packages/nuxt-module/plugins/domain.js#L175 

One of those is undefined, so it would be a better fix in my opinion to add a condition to check the existance of `domainConfig` and `fromDomainConfig`:
1) https://github.com/vuestorefront/shopware-pwa/blob/258e6d51193ff2f8eae70b53af62d265fd3e4141/packages/nuxt-module/plugins/domain.js#L157
2) https://github.com/vuestorefront/shopware-pwa/blob/258e6d51193ff2f8eae70b53af62d265fd3e4141/packages/nuxt-module/plugins/domain.js#L181 

So it would be `domainConfig && !hasPrefixPath(domainConfig)` and `fromDomainConfig && !hasPrefixPath(fromDomainConfig)`

Additionally the optionalChaining can be added in `hasPrefixPath`. 
I'd rather not add the condition, because this is not what the function is about: it shouldn't return `false`, when `domainConfig` isn't set, this would be misleading. It should throw an exception instead.

Reply:
I found that the error probably comes from the point 2. (https://github.com/vuestorefront/shopware-pwa/blob/258e6d51193ff2f8eae70b53af62d265fd3e4141/packages/nuxt-module/plugins/domain.js#L175) mentioned by Niklas and it's because that on SSR there is now `from object set and `fromDomainConfig` is always `undefined`. @mmeester could you add another condition for that case?